src/gc/gc.d

```diff
@@ -605,61 +605,72 @@ struct GC
             else
             {
                 auto pool = gcx.findPool(p);
-                psize = pool.getSize(p);     // get allocated size
-                if (psize >= PAGESIZE && size >= PAGESIZE)
+                if (pool.isLargeObject)
```

Comment:
Agreed. I was tempted to move part of this into Gcx, but resisted to not make reviewing worse.